magda-sleuther-format-layerer/src/format-engine/measures/dcatFormatMeasure.ts

```diff
@@ -7,106 +7,103 @@ import { Record } from "@magda/typescript-common/dist/generated/registry/api";
+            format.indexOf("(") === format.indexOf(")")
+        );
+    });
 }
 
 export default function getMeasureResult(
-    relatedDistribution: Record
+    relatedDistribution: Record,
+    synonymObject: any
 ): MeasureResult {
     const { format } = relatedDistribution.aspects["dcat-distribution-strings"];
 
     if (format === null || format === "") {
         return null;
     }
 
-    let processedFormat: string;
-
-    const transforms = [
-        (str: string) =>
-            lowerCase(
-                removePeriod(
-                    replaceComma(
-                        replaceSpaceThenComma(replaceCommaThenSpace(str))
-                    )
-                )
-            )
+    // this is an array that acts like an assembly belt, you input in the string, the middle functions are the assembly robots,
+    // and the last function returns the output.
+    const cleanUpAssemblyChain = [
+        format,
+        foundationalCleanup,
+        replaceAmpersandFormats,
+        splitWhiteSpaceFormats,
+        reduceMimeType,
+        filterBracketedFormats
     ];
 
-    for (const tf of transforms) {
-        processedFormat = applyTransform(format, tf);
-    }
-
-    // hard coded rules for separating out multiple formats when provided
-    let splitFormat: Array<string> = [];
-    let finalFormat: Array<string> = [];
-
-    // e.g: application/exe
-    splitFormat = processedFormat.split("/");
-    if (splitFormat.length > 2)
-        throw new Error(
-            "a MIME type has more than 1 slash: " + processedFormat
-        );
-    if (splitFormat.length > 1) finalFormat.push(splitFormat[1]);
-    else {
-        // E.g. pdf & xlsx & doc & docx & ogg
-        splitFormat = processedFormat.split(" & ");
-        if (splitFormat.length > 1) finalFormat = splitFormat;
-        else {
-            splitFormat = processedFormat.split(" ");
-            if (splitFormat.length > 1) {
-                // E.g. zip (xlsx)
-                if (processedFormat.indexOf("(") > -1) {
-                    //TODO make this more efficient or elegant
-                    let length: number =
-                        processedFormat.substr(processedFormat.indexOf(")"))
-                            .length -
-                        processedFormat.substr(processedFormat.indexOf("("))
-                            .length;
-                    finalFormat.push(processedFormat.substr(
-                        processedFormat.indexOf("("),
-                        length
-                    ));
-                } else {
-                    // E.g. ATOM
-                    finalFormat.push(splitFormat[0]);
-                }
-            } else {
-                console.log("f:" + finalFormat);
-                // can only deduce 1 format in this scenario
-                finalFormat.push(processedFormat);
-            }
-        }
-    }
+    const processedFormat: Array<string> = cleanUpAssemblyChain.reduce(function(
```

Comment:
I really like this assembly chain idea in general but I think it'd be a bit clearer if you made it so that `cleanUpAssemblyChain` was all functions, and`format` was the last argument of reduce.

Like

```
cleanUpAssemblyChain.reduce(function(
    accumliation,
    currentObject
) {
    return currentObject(accumliation);
}, format)
```

Should still work the same but there won't be any mixed types in cleanUpAssemblyChain.